### DISCUSSION TITLE
How to use the dark mode for the new homepage scroll bar?

### DISCUSSION THREAD

**Discussion by ouweiya** [tailwindlabs/tailwindcss#6734] [General]
I see that the scroll bar is displayed well in dark mode.
To achieve this effect, is the pseudo element selector used?

`::-webkit-scrollbar`,`::-webkit-scrollbar-thumb`

![image](https://user-images.githubusercontent.com/28357731/147390148-15aed9cb-0c7a-496e-b5eb-ec4e0809ef8b.png)

**Comment by enjidev** [comment]
Hai :wave: you can try this one: 

```CSS
.dark {
  color-scheme: dark;
}
```

**Reply by ouweiya** [reply]
Thank you for sharing. This is a very cool attribute. I see that V3 introduces a lot of very new CSS APIs, which I like very much.

**Reply by ouweiya** [reply]
Simple mode
`dark:[color-scheme:dark]`

**Reply by enjidev** [reply]
ah, that's better!

**Reply by rutikwankhade** [reply]
can you please give an example of how I can use it with scrollbars?

**Reply by ouweiya** [reply]
Add `dark:[color-scheme:dark]` to any element class name that produces scrolling

**Reply by lukewarlow** [reply]
I would recommend doing [color-scheme:light_dark] instead. That tells the browser you support both light and dark and will switch when the setting changes. Will (I think) avoid issues in Safari with dynamic color-scheme properties not updating scrollbars properly (fwiw I have fixed this so next Safari release should work correctly)

**Comment by Drammy** [comment]
I find that`dark:[color-scheme:dark]` breaks when using `darkMode: 'media'` and I toggle dark mode back and forth on the OS.

Anyone else find this?  If not, can you please help..?


edit: I've only tested this on safari on Mac so far.

**Reply by Drammy** [reply]
A bug with safari perhaps?

**Reply by lukewarlow** [reply]
When you say breaks what do you mean?

Something you could do instead is something like [color-scheme:dark_light] which shows that you support both dark and light mode and it will match the media query. So dark scrollbar in dark mode and light scrollbar in light mode.

**Reply by Drammy** [reply]
Apologies for the rather useless description of the issue..

By broken I mean the scrollbar doesn't change from dark to light mode and vice versa unless the user hovers over the scrollbar..
![image](https://github.com/tailwindlabs/tailwindcss/assets/5029037/60680704-7f75-4789-b34a-ea1a0311dce0)

**Reply by lukewarlow** [reply]
That almost feels like a browser bug rather than a TailwindCSS one. Which browser are you using? Because I know Safari has issues with dynamic scrollbar styling.

**Reply by Drammy** [reply]
Yeah, I questioned above if its just a safari bug; that's what I put it down to anyway.
Just thought I'd mention it...

**Reply by lukewarlow** [reply]
I'll take a look and see if I can make a minimal reproduction in safari to report it there as a bug.

**Reply by lukewarlow** [reply]
Just a heads up here's the safari bug: https://bugs.webkit.org/show_bug.cgi?id=244494

I've submitted a PR to the WebKit repository that should fix this.

**Reply by Drammy** [reply]
That's great, thanks for the update!

**Reply by lukewarlow** [reply]
That bug fix has been merged so 🤞 the next Safari release (or potentially the one after that) will include the bug fix.

**Reply by thexmep** [reply]
In reaction I have solved it like this.  in css
.dark::-webkit-scrollbar {
  background-color: #333; 
}

.dark::-webkit-scrollbar-thumb {
  background-color: #555; 
}
